framework.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import platform
+
+import sgtk
+logger = sgtk.platform.get_logger(__name__)
+
+
+def get_vendors_path():
+    """Return the path to the vendors folder.
+
+    The vendors folder is a way to package the dependecies of this framework in a "It just works" manner.
+
+    :returns: path to the vendors folder
+    :rtype: str
+    """
+    def get_python_version():
+        if sys.hexversion < 0x3000000:
+            return "2"
+
+        return "3"
```

Comment:
Did you try your code with Will's in-progress port of tk-core to Python 3? No rush to fix it if it doesn't. Will, Jeff and I will go through everything in the coming weeks.

Reply:
I didn't

Reply:
No problem waiting on us to do a Python 3 pass alongside you late on, @geffrak. If it turns out there are compatibility problems we can sort those out as we progress in the Python 3 project.